certification/internal/engine/openshift.go

```diff
@@ -15,15 +16,31 @@ import (
 	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/cli"
 	client "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/client"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
-type OpenshiftEngine struct {
+type openshiftEngine struct {
 	KubeConfig *rest.Config
 }
 
-func (pe OpenshiftEngine) CreateNamespace(name string, opts cli.OpenshiftOptions) (*cli.OpenshiftReport, error) {
+func NewOpenshiftEngine() *cli.OpenshiftEngine {
+	var engine cli.OpenshiftEngine = &openshiftEngine{}
+	return &engine
+}
+
+func (oe *openshiftEngine) Setup() error {
+	k8sconfig, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		log.Error("unable to parse kubecofig: ", err)
+		return err
+	}
+	oe.KubeConfig = k8sconfig
+	return nil
+}
+
+func (oe *openshiftEngine) CreateNamespace(name string, opts cli.OpenshiftOptions) (*corev1.Namespace, error) {
```

Comment:
Given that `Setup()` doesn't take any parameters, would it make sense to just perform the tasks in that method in the `NewOpenshiftEngine`?

Reply:
I initially did this. But it causes the tests in engine/ folder to fail because `KUBECONFIG` has not been set and `clientcmd.BuildConfigFromFlags` returns nil dereference (engine.go registers and initializes all checks). The way it is done now, lets us to mock the call to Setup(). I will be glad to revert it back to what it was before if there is any way to mock this.